http-client/Network/HTTP/Client/Core.hs

```diff
@@ -69,11 +70,12 @@ httpLbs req man = withResponse req man $ \res -> do
 httpNoBody :: Request -> Manager -> IO (Response ())
 httpNoBody req man = withResponse req man $ return . void
 
+
 -- | Get a 'Response' without any redirect following.
 httpRaw
      :: Request
      -> Manager
-     -> IO (Response BodyReader)
+     -> IO (Request, Response BodyReader)
```

Comment:
I'd misunderstood your previous comment, I thought you had intended to change the behavior of `httpRaw` but keep the type signature the same. Given this, making a new function with this type signature would be a better approach.